src/main/java/com/alkemy/ong/ports/input/rs/controller/UserController.java

```diff
@@ -31,4 +41,16 @@ public ResponseEntity<List<UserResponse>> getUsers() {
         return ResponseEntity.ok().body(response);
     }
 
+    @PatchMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void updateUser(@NotNull @PathVariable Long id, @Valid @RequestBody UpdateUserRequest updateUserRequest, @AuthenticationPrincipal User loggedUser) {
+        System.out.println(loggedUser.getRole().getName());
+        if (loggedUser.getId() == id || loggedUser.getRole().getName().equals("ROLE_ADMIN")) {
+            User user = mapper.updateUserRequestToUser(updateUserRequest);
+            service.updateEntityIfExists(id, user);
+        } else {
+            throw new BadCredentialsException("");
```

Comment:
Debería lanzar `throw new AccessDeniedException("Access denied to resource");`

Reply:
Listo!